Problem 1.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <numeric>
+
+using namespace std;
+
+int main()
+{
+	vector<string> a;
+	string p;
+	do {
+		getline(cin, p);
+		a.push_back(p);
+	} while (p != "");
+	a.pop_back();
+
+	if (all_of(a.begin(), a.end(), [](string& i) {return (i.size() >= 4); })) {
+		cout << "ALL\n";
+	}
+	else if (any_of(a.begin(), a.end(), [](string& i) {return (i.size() >= 4); })) {
+		cout << "SOME\n";
+	}
+	else if (none_of(a.begin(), a.end(), [](string& i) {return (i.size() >= 4); })) {
+		cout << "NONE\n";
+	}
+
+	for_each(a.begin(), a.end(), [](string& i) {cout << i.back() << "\n"; });
+
+	cout << count_if(a.begin(), a.end(), [](string& i) {return count_if(i.begin(), i.end(), isdigit); }) << "\n";
+
+	transform(a.begin(), a.end(), a.begin(), [](string& i) {transform(i.begin(), i.end(), i.begin(), tolower); return NULL; });
+
+	vector<pair<int, int>> pa(a.size());
+	int i = 0;
+	transform(a.begin(), a.end(), pa.begin(), [&i](string& p) {pair<int, int> s; s.first = accumulate(p.begin(), p.end(), 0);  s.second = i; i++; return s; });
```

Comment:
The logic is fine, but you should format your lambda expressions to look more like real functions. Put one statement per line.

Reply:
Why is this better? It looks much better all on one line.

Reply:
First of all, no it doesn't. Second of all, it's not as readable. I have to go through and parse out where each statement begins/ends to understand what's written here. Here is what I see when I review the code:
![image](https://user-images.githubusercontent.com/5101039/74082745-f8f61e00-4a11-11ea-8d06-3e37ca8275ba.png)

I suggest formatting it like this:
```
transform(a.begin(), a.end(), pa.begin(), [&i](string p)
{
        pair<int, int> s; 
        s.first = accumulate(p.begin(), p.end(), 0);  
        s.second = i;
        i++; 
        return s; 
}); 
```
This is similar formatting to when you define a function, and is much more readable. 
In general, at tech companies your review scores are based on how much code you merge. The easier you make life for your reviewers, the more code you'll get merged, the more money you'll make.